operators/src/processing/expression.rs

```diff
@@ -38,39 +37,11 @@ use std::marker::PhantomData;
 pub struct ExpressionParams {
     pub expression: String,
     pub output_type: RasterDataType,
-    #[serde(deserialize_with = "parse_no_data")]
-    #[serde(serialize_with = "write_no_data")]
+    #[serde(with = "float_with_nan")]
```

Comment:
what was wrong with the old approach? Did it ever work?

Reply:
The old approach used the JSON parser internally. But Serde isn't all JSON. But other than that it should work the same.